src/background/services/lock/index.ts

```diff
@@ -99,27 +103,58 @@ export default class LockService implements IBackupable {
 
   downloadEncryptedStorage = (): Promise<string | null> => this.passwordStorage.get<string>();
 
-  uploadEncryptedStorage = async (encryptedStorage: string, password: string): Promise<void> => {
-    await this.checkPassword(password);
-    await this.passwordStorage.set(encryptedStorage);
+  uploadEncryptedStorage = async (encryptedBackup: string, password: string): Promise<void> => {
+    const { isNew, authenticBackupCiphertext } = await this.isAuthenticated(password, encryptedBackup);
+    if (isNew && authenticBackupCiphertext) await this.passwordStorage.set(authenticBackupCiphertext);
   };
 
-  checkPassword = async (password: string): Promise<void> => {
+  // TODO: Unit test needed, it is meant to replace checkPassword
+  isAuthenticated = async (
+    password: string,
+    backupCiphertext?: string,
+  ): Promise<{ isAuthentic: boolean; isBackup?: boolean; isNew?: boolean; authenticBackupCiphertext?: string }> => {
     const cipherText = await this.passwordStorage.get<string>();
 
-    if (!cipherText) {
-      throw new Error("Something badly gone wrong (reinstallation probably required)");
+    if (!password) throw new Error("Password is not provided");
+
+    // Normal check
+    if (cipherText) {
+      const decryptedPasswordChecker = cryptoDecrypt(cipherText, password);
+
+      if (decryptedPasswordChecker !== this.passwordChecker) throw new Error("Incorrect password");
+      else {
+        if (backupCiphertext)
+          return {
+            isAuthentic: true,
+            isNew: false,
+            authenticBackupCiphertext: this.isBackupCiphertextAutentic(backupCiphertext),
+          };
+
+        return {
+          isAuthentic: true,
+        };
+      }
     }
+    // Newly installed CK
+    // We should return an object if it
+    else {
+      if (backupCiphertext) {
+        return {
+          isAuthentic: true, // Becuase a new installed wallet
+          isNew: true,
+          authenticBackupCiphertext: this.isBackupCiphertextAutentic(backupCiphertext),
+        };
+      }
 
-    if (!password) {
-      throw new Error("Password is not provided");
+      throw new Error("Something badly gone wrong (reinstallation probably required)");
     }
+  };
 
-    const bytes = CryptoJS.AES.decrypt(cipherText, password);
-    const retrievedPasswordChecker = bytes.toString(CryptoJS.enc.Utf8);
-
-    if (retrievedPasswordChecker !== this.passwordChecker) {
-      throw new Error("Incorrect password");
+  private isBackupCiphertextAutentic = (backupCiphertext: string): string => {
```

Comment:
Methods which starts with `is` is supposed to return boolean value. I'd rename it to `getBackupCiphertextAutentic`